web/src/engine/osk/src/views/oskView.ts

```diff
@@ -595,8 +596,33 @@ export default abstract class OSKView
     this.needsLayout = true;
   }
 
+  public batchLayoutAfter(closure: () => void) {
+    /*
+      Is there already an ongoing batch?  If so, just run the closure and don't
+      adjust the tracking variables.  The outermost call will finalize layout.
+    */
+    if(this.deferLayout) {
+      closure();
+      return;
+    }
+
+    try {
+      this.deferLayout = true;
+      if(this.vkbd) {
+        this.vkbd.deferLayout = true;
+      }
+      closure();
+    } finally {
+      this.deferLayout = false;
+      if(this.vkbd) {
+        this.vkbd.deferLayout = false;
+      }
+      this.refreshLayout();
+    }
```

Comment:
Are these the only places that deferLayout variables are set?

You could probably have done `this.vkbd?.deferLayout = false;` for shorter code.

Reply:
```
The left-hand side of an assignment expression may not be an optional property access.ts(2779)
```

TS does not allow that "shorter code" variant, unfortunately.  I've tried to use it a few times.

Reply:
Within `OSKView`, it does appear so.

Reply:
Note that it is possible for desktop keyboards to use a help-page structure (like `sil_euro_latin`), in which case we can't use this.vkbd's version of the field.